src/engine/Match_env.ml

```diff
@@ -34,6 +34,10 @@ type pattern_id = Xpattern.pattern_id
 (* !This hash table uses the Hashtbl.find_all property! *)
 type id_to_match_results = (pattern_id, Pattern_match.t) Hashtbl.t
 
+type filter_rules_config =
```

Comment:
maybe prefilter_config instead?
and then
| PrefilterCache of ...
| NoPrefiltering